### ISSUE TITLE
.isValidated() returns true even if there's an error

### ISSUE THREAD

**Issue by ksami** [voltrue2/in-app-purchase#119]
The `.isValidated()` method only checks if the response from `.validate()` has `status === 0` and returns `true` based on that.  
However if for example, deleting a character from the purchase token, `status` is still `0` so the `isValidated()` method still returns `true`.

Example response:
```
{
    "packageName": ...,
    "productId": ...,
    "purchaseTime": 1503996911431,
    "purchaseState": 0,
    "purchaseToken": valid purchaseToken with one character missing,
    "status": 0,
    "error": {
        "errors": [{
            "domain": "global",
            "reason": "invalid",
            "message": "Invalid Value"
        }],
        "code": 400,
        "message": "Invalid Value"
    },
    "service": "google"
}
```

**Comment by voltrue2** [comment]
Hello

`.isValidated()` is a support function that is meant to be a support function for `.validate()` and `.validateOnce()`.
This function does not validate on its own.

Let me know if this answer clarifies your question or not.

Cheers

**Comment by ksami** [comment]
Hi @voltrue2 ,

Yup I understand that. But the steps I took are:

1. Delete one character from purchaseToken of a valid receipt
2. Call `.validate()` with the edited receipt to get the response
3. Call `.isValidated()` with the response from `.validate()`
4. `.isValidated()` returns `true` instead of `false`

**Comment by voltrue2** [comment]
@ksami 
I see. Thank you for reporting this. I will have a look at this issue.

Cheers!

**Comment by voltrue2** [comment]
@ksami 
I have implemented the fixed for this issue.

The problem as I understand it comes from the actual validation that it does not check the HTTP response status code for GooglePlay API.
I believe you are using `googleRefToken`, `googleClientID`, and `googleClientSecret` instead of public key, correct?

Here is the the fixed code: d94b674c8feb0c698e5e1151a52dc3d41cd57e5c

Please let me know if this fixes the problem you are facing.

Cheers!

**Comment by ksami** [comment]
@voltrue2 
Wow thanks for fixing it so quickly! Yes I was using `googleRefToken`, `googleClientID` and `googleClientSecret`.

d94b674 does fix this problem. Thanks again for the work you put into this!
